### ISSUE TITLE
RustCrypto KEM and Signature trait support

### ISSUE THREAD

**Issue by thomwiggers** [rustpq/pqcrypto#40]
FYI I just cut a new version of [`pcrypto-compat`](https://github.com/rozbb/pqcrypto-compat) which implements the RustCrypto KEM and signature traits over the `pqcrypto` primitives. If it looks like a nice interface, I could make a PR here that implements it directly, and deprecate my crate. Let me know!

_Originally posted by @rozbb in https://github.com/rustpq/pqcrypto/issues/39#issuecomment-1139385517_

**Comment by thomwiggers** [comment]
Supporting RustCrypto's traits is very much something that I want to have. I don't think I'll have time before July to work on this myself (I'm basically AFK after next Wednesday and have some deadlines before) -- but I will definitely welcome any PR (though I won't be able to review anything in June.

I'll also create an issue on the OQS Rust bindings repository.

**Comment by rozbb** [comment]
Awesome! This'll be a little complicated because of all the codegen, but I'll take a crack at it. I'm also quite busy but this summer is a good time for me overall

**Comment by thomwiggers** [comment]
Yeah I do apologize for that; it's grown to be a bit of a beast 😅

**Comment by mouse07410** [comment]
Any progress since...?
